app/src/main/java/com/grupo3/vinilos/artists/detail/ArtistDetailViewModel.kt

```diff
@@ -4,40 +4,36 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.grupo3.vinilos.artists.service.ArtistsRepository
 import com.grupo3.vinilos.utils.ERROR_MESSAGE
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
-import java.lang.Exception
 
 class ArtistDetailViewModel : ViewModel() {
     private val repository = ArtistsRepository()
     private val _state = MutableStateFlow(ArtistDetailState())
     val state: StateFlow<ArtistDetailState> = _state.asStateFlow()
-    suspend fun getArtist(id: Int) {
-        try {
-            val artist = repository.getMusician(id)
+    fun getArtist(id: Int) {
+        viewModelScope.launch(Dispatchers.IO) {
+            val musician = try {
+                repository.getMusician(id)
+            } catch (e: Exception) {
+                null
+            }
+            val band = try {
+                repository.getBand(id)
```

Comment:
Should they can be moved to the same try-catch?

Reply:
No, what if the id that was sent is from a band? 
if so the musician request will throw and error, and the get band method won't be executed.